src/Diagnostics.HealthChecks/OmexHealthCheckPublisherHostedService.cs

```diff
@@ -0,0 +1,351 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#pragma warning disable IDE1006 // ASP.NET Core codebase naming convention
+#pragma warning disable IDE0008 // Use explicit type
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Internal;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks;
+
+internal sealed partial class OmexHealthCheckPublisherHostedService : IHostedService
```

Comment:
This class has a weird indentation and would be worth adding a summary as this is the entry point for the health checks?